lib/pbench/test/unit/server/conftest.py

```diff
@@ -736,31 +700,131 @@ def create_drb_user(client, server_config, fake_email_validator):
 
 
 @pytest.fixture()
-def pbench_token(client, server_config, create_drb_user):
-    # Login user to get valid pbench token
-    response = login_user(client, server_config, "drb", generic_password)
-    assert response.status_code == HTTPStatus.OK
-    data = response.json
-    assert data["auth_token"]
-    return data["auth_token"]
+def pbench_admin_token(client, create_admin_user):
+    return generate_token(
+        user=create_admin_user,
+        username=admin_username,
+        pbench_client_roles=["ADMIN"],
+    )
+
+
+@pytest.fixture()
+def pbench_token(client, create_drb_user):
+    """
+    OIDC valid token for the 'drb' user.
+    """
+    return generate_token(username="drb", user=create_drb_user)
+
+
+@pytest.fixture()
+def pbench_token_invalid(client, create_drb_user):
+    """
+    OIDC invalid token for the 'drb' user
+    """
+    return generate_token(username="drb", user=create_drb_user, valid=False)
+
+
+def generate_token(
+    username: str,
+    user: Optional[User] = None,
+    pbench_client_roles: Optional[list[str]] = None,
+    valid: Optional[bool] = True,
+) -> str:
+    """
+    Generates an OIDC JWT token that mimics the real OIDC token obtained
+    from the OIDC complaint client login.
+
+    Args:
+        username: username to include in the token payload
+        user: user attributes will be extracted from the user to include in
+            the token payload, if not provided it will be queried from the User
+            table. It is a responsibility of the consumer to make sure the
+            user exists in the 'User' table until we remove the User table
+            completely.
```

Comment:
Instead of a comma, that should probably be a period and a new sentence.

In that new sentence, the "it" is a little ambiguous...I think what you're trying to say is that the record will be fetched from the `User` table using the `username` as a key.

At line 741, I think "caller" probably works better than "consumer".